### ISSUE TITLE
Inconsistent concatenate behaviour

### ISSUE THREAD

**Issue by pbazant** [numpy/numpy#5759]
Numpy under Python 2.7.9:

```
In [10]: np.version.version
Out[10]: '1.9.1'

In [11]: np.concatenate(([[1,2]],[[3,4]]), axis = 10)
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-11-2a6d0286a78f> in <module>()
----> 1 np.concatenate(([[1,2]],[[3,4]]), axis = 10)

IndexError: axis 10 out of bounds [0, 2)

In [12]: np.concatenate(([1,2],[3,4]), axis = 10)
Out[12]: array([1, 2, 3, 4])
```

I would expect the second use of concatenate to throw an exception, too. Is this behaviour intentional?

**Comment by jaimefrio** [comment]
There is a deprecation warning in place, try this:

```
>>> import warnings
>>> warnings.simplefilter('always')
>>> np.concatenate(([1,2],[3,4]), axis = 10)
__main__:1: DeprecationWarning: axis != 0 for ndim == 1; this will raise an error in future versions of numpy
array([1, 2, 3, 4])
```

It was introduced in 21a1d73 and 4475ead, almost 3 years ago. Perhaps 1.10 could be the deprecated future?

**Comment by njsmith** [comment]
I guess we should look up when the first NumPy release containing those
commits was instead of just when they were written, but assuming that's
also 2+ years ago then raising an error in 1.10 seems reasonable to me.

**Comment by jaimefrio** [comment]
[This comment](https://github.com/numpy/numpy/blob/4cbced22a8306d7214a4e18d12e651c034143922/numpy/core/tests/test_shape_base.py#L235) seems to indicate it went into effect in 1.7.

**Comment by ahaldane** [comment]
Fixed by #8584. Closing.
